tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java

```diff
@@ -146,6 +146,57 @@ public String getValue() {
     }
   }
 
+  @Test public void exceptionsFromJavaWithUnifiedStackTrace() {
+    TestInterface boundObject = new TestInterface() {
+      @Override public String getValue() {
+        throw new UnsupportedOperationException("Cannot getValue");
+      }
+    };
+    duktape.bind("value", TestInterface.class, boundObject);
+    try {
+      duktape.evaluate("\n"
+          + "f1();\n"           // Line 2.
+          + "\n"
+          + "function f1() {\n"
+          + "  f2();\n"         // Line 5.
+          + "}\n"
+          + "\n"
+          + "\n"
+          + "function f2() {\n"
+          + "  value.getValue();\n"       // Line 10.
+          + "}\n", "test.js");
+      fail();
+    } catch (UnsupportedOperationException e) {
+      assertThat(e).hasMessage("Cannot getValue");
+
+      StackTraceElement[] stackTrace = e.getStackTrace();
+
+      // Top entry is what threw - the TestInterface implementation.
+      assertThat(stackTrace[0].getClassName()).isEqualTo(boundObject.getClass().getName());
```

Comment:
I had initially done that, and ran into just that.  There's also the ugliness/brittleness of the line numbers that show up in the printed stack trace.